tinywav.c

```diff
@@ -191,23 +198,23 @@ size_t tinywav_write_f(TinyWav *tw, void *f, int len) {
           const float **const x = (const float **const) f;
           for (int i = 0, k = 0; i < len; ++i) {
             for (int j = 0; j < tw->numChannels; ++j) {
-              z[k++] = (int16_t) (x[j][i] * 32767.0f);
+              z[k++] = (int16_t) (x[j][i] * INT16_MAX);
             }
           }
           break;
         }
         default: return 0;
       }
 
-      tw->totalFramesWritten += len;
-      return fwrite(z, sizeof(int16_t), tw->numChannels*len, tw->f);
-      break;
+      tw->totalFramesReadWritten += len;
+      samples_written = fwrite(z, sizeof(int16_t), tw->numChannels*len, tw->f);
+      return (int) samples_written / tw->numChannels;
     }
     case TW_FLOAT32: {
       float *z = (float *) alloca(tw->numChannels*len*sizeof(float));
       switch (tw->chanFmt) {
         case TW_INTERLEAVED: {
-          tw->totalFramesWritten += len;
+          tw->totalFramesReadWritten += len;
           return fwrite(f, sizeof(float), tw->numChannels*len, tw->f);
```

Comment:
Should we be accounting here also for the number of channels?

Reply:
The way I saw it throughout tinywav, a frame is defined as

`1 frame = [1 sample for each channel]`

The number of samples is equal for a channels, so implicitly, `numTotalSamples = numFrames * numChannels`

Reply:
Ah, or did you mean we should add a member to track the number of channels written/read? That wouldn't make any sense IMO, because there's no way to read/write channels individually.